File: MomentsJava/src/main/java/com/example/momentsjava/data/api/ListApiService.java
package com.example.momentsjava.data.api;

import com.example.momentsjava.data.model.ListItem;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;

public interface ListApiService {
    @GET("list")
    Observable<List<ListItem>> getList();
}

File: app/src/main/java/com/example/simplelist/network/ListApiService.kt
package com.example.simplelist.network

import com.example.simplelist.model.ListItem
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET

private const val BASE_URL = "http://10.0.2.2:3022"

private val retrofit = Retrofit.Builder()
    .addConverterFactory(GsonConverterFactory.create())
    .baseUrl(BASE_URL)
    .build()

interface ListApiService{
    @GET("list")
    suspend fun getList():List<ListItem>
}

object ListApi{
    val retrofitService: ListApiService by lazy {
        retrofit.create(ListApiService::class.java)
    }
}
